app/uk/gov/hmrc/helptosave/connectors/BarsConnector.scala

```diff
@@ -36,18 +37,19 @@ trait BarsConnector {
 }
 
 @Singleton
-class BarsConnectorImpl @Inject()(http: HttpClient)(implicit appConfig: AppConfig) extends BarsConnector with Logging {
+class BarsConnectorImpl @Inject()(http: HttpClientV2)(implicit appConfig: AppConfig) extends BarsConnector with Logging {
 
   import uk.gov.hmrc.helptosave.connectors.BarsConnectorImpl._
 
-  private val barsEndpoint: String = s"${appConfig.barsUrl}/validate/bank-details"
+  private val barsEndpoint: URL = url"${appConfig.barsUrl}/validate/bank-details"
 
-  private val headers = Map("Content-Type" -> "application/json")
+  private val headers:(String, String) = "Content-Type" -> "application/json"
 
   override def validate(request: BankDetailsValidationRequest, trackingId: UUID)(
     implicit hc: HeaderCarrier,
-    ec: ExecutionContext): Future[HttpResponse] =
-    http.post(barsEndpoint, bodyJson(request), headers.+("X-Tracking-Id" -> trackingId.toString))
+    ec: ExecutionContext): Future[HttpResponse] = {
+    http.post(barsEndpoint).withBody(bodyJson(request)).transform(_.addHttpHeaders(headers, "X-Tracking-Id" -> trackingId.toString)).execute[HttpResponse]
```

Comment:
Sorry, missed this in previous review. 
Would it also make sense to capture the return type as `Either[UpstreamErrorResponse, HttpResponse]` here? This might need tests being tidied up to reflect the change

While doing that, could you also check on other connectors making sure of the return type and error handling in calling code.